Assets/UI/worldinput_CQUI.lua

```diff
@@ -0,0 +1,768 @@
+  if UI.GetInterfaceMode() == InterfaceModeTypes.DISTRICT_RANGE_ATTACK then
+    UI.SetInterfaceMode(InterfaceModeTypes.SELECTION);
+  end
+
+  return BASE_CQUI_OnUnitSelectionChanged( playerID, unitID, hexI, hexJ, hexK, isSelected, isEditable );
+end
+
+-- ===========================================================================
+function DefaultKeyDownHandler( uiKey:number )
+  print_debug("** Function Entry: DefaultKeyDownHandler (CQUI Hook).  uiKey: "..tostring(uiKey));
+  -- Note: This function always returns false, by design in Base game
+  BASE_CQUI_DefaultKeyDownHandler( uiKey );
+
+  --CQUI Keybinds
+  local keyPanChanged :boolean = false;
+
+  if uiKey == Keys.VK_SHIFT then
+    print_debug("CQUI_isShiftDown = true");
+    CQUI_isShiftDown = true;
+  end
+
+  if uiKey == Keys.VK_ALT then
+    print_debug("CQUI_isAltDown = true");
+    CQUI_isAltDown = true;
+  end
+
+  if CQUI_hotkeyMode == CQUI_HOTKEYMODE_ENHANCED then
+    if (uiKey == Keys.W) then
+      keyPanChanged = true;
+      m_isUPpressed = true;
+    end
+
+    if (uiKey == Keys.D) then
+      keyPanChanged = true;
+      m_isRIGHTpressed = true;
+    end
+
+    if (uiKey == Keys.S) then
+      keyPanChanged = true;
+      m_isDOWNpressed = true;
+    end
+
+    if (uiKey == Keys.A) then
+      keyPanChanged = true;
+      m_isLEFTpressed = true;
+    end
+  end
+
+  if (keyPanChanged == true) then
+    -- Base game file uses m_edgePanX and m_edgePanY... but those values do not ever appear change from 0, so just send 0.
+    ProcessPan(0, 0);
+    return true;
+  end
+
+  -- Allow other handlers of KeyDown events to capture input, if any exist after World Input
+  return false;
+end
+
+-- ===========================================================================
+function DefaultKeyUpHandler( uiKey:number )
+  print_debug("** Function Entry: DefaultKeyUpHandler (CQUI Hook) uiKey: "..tostring(uiKey));
+
+  -- If set to Standard, just let the game do its thing
+  if CQUI_hotkeyMode == CQUI_HOTKEYMODE_STANDARD then
+    if uiKey == Keys.VK_SHIFT then
+      CQUI_isShiftDown = false;
+    end
+  
+    if uiKey == Keys.VK_ALT then
+      CQUI_isAltDown = false;
+    end
+
+    return BASE_CQUI_DefaultKeyUpHandler(uiKey);
+  end
+
+  local keyPanChanged  :boolean = false;
+  local cquiHandledKey :boolean = false;
+
+  local selectedUnit = UI.GetHeadSelectedUnit();
+  local unitType = nil;
+  local formationClass = nil;
+  if (selectedUnit) then
+    unitType = GameInfo.Units[selectedUnit:GetUnitType()].UnitType;
+    formationClass = GameInfo.Units[selectedUnit:GetUnitType()].FormationClass;
+  end
+
+  --CQUI Keybinds
+  if CQUI_hotkeyMode == CQUI_HOTKEYMODE_ENHANCED then
+    if (uiKey == Keys.W) then
+      m_isUPpressed = false;
+      keyPanChanged = true;
+    end
+
+    if (uiKey == Keys.D) then
+      m_isRIGHTpressed = false;
+      keyPanChanged = true;
+    end
+
+    if (uiKey == Keys.S and CQUI_isAltDown == false) then
```

Comment:
I've tried it with this change and I think the old code was better. With the current code you can get a situation where you press Alt, then press S (key down handler starts moving it down now), then release S (the new code will not clear the flag `m_isDOWNpressed` since Alt is still held down), then release Alt. The game will just keep panning down even with no keys held, I was thinking it should stop moving down when Alt is released but for some reason it doesn't.

I think a better change would be to modify the air strike command to something else. Or just comment out or remove since it doesn't seem to work anyway.

Reply:
I think you're right.. you mentioned below - the strike and the nuke commands require that we're pointing at a unit or tile and by itself... which I guess you could do if the mouse is pointing at it?
But even then... maybe just comment it out with a TODO, figure it out kind of thing

Reply:
Yup, makes sense to comment out the air commands. Just don't forget to change this back to 
```suggestion
    if (uiKey == Keys.S) then
```
Otherwise we'll have a bug if someone tries to press Alt+S